### PR TITLE
Liquidation can be non-fraud

### PR DESCRIPTION
**Pull request by NicholasDotSol**
closes: #497

`LIQUIDATION_IN_PROGRESS` was never used.

set state to `FRAUD_LIQUIDATION IN PROGRESS` when fraud,
and `LIQUIDATION_IN_PROGRESS` when not fraud.

### CODE REVIEW

File: implementation/contracts/deposit/DepositLiquidation.sol [hunk 1]
```diff
@@ -95,7 +95,7 @@ library DepositLiquidation {
 
         _d.liquidationInitiated = block.timestamp;  // Store the timestamp for auction
         _d.liquidationInitiator = msg.sender;
-        _d.setFraudLiquidationInProgress();
+        _d.setLiquidationInProgress();
```
Comment (nkuba) [inline implementation/contracts/deposit/DepositLiquidation.sol]: Doesn't this change require an update in unit tests? Can we make sure we verify state in tests after calling this function (for `notifyUndercollateralizedLiquidation` and `notifyCourtesyTimeout`)?

Comment (NicholasDotSol) [reply implementation/contracts/deposit/DepositLiquidation.sol]: > Doesn't this change require an update in unit tests?

nope, we didn't verify state before.

> Can we make sure we verify state in tests after calling this function

Yes! adding state verification for `notifyUndercollateralizedLiquidation` `notifyCourtesyTimeout` `provideECDSAFraudProof` and `provideSPVFraudProof`

**Comment by nkuba** [comment]
@NicholasDotSol LGTM. Can we just update this branch with master?

**Comment by NicholasDotSol** [comment]
@nkuba ready here

**Pull request merged**